be/src/storage/vectorized/compaction.h

```diff
@@ -52,24 +72,31 @@ class Compaction {
     Status do_compaction();
     Status do_compaction_impl();
 
-    // merge rows from vectorized reader and write into `_output_rs_writer`.
-    // return Status::OK() and set statistics into `*stats_output`.
-    // return others on error
-    Status merge_rowsets(int64_t mem_limit, Statistics* stats_output);
-
     Status modify_rowsets();
 
-    Status construct_output_rowset_writer();
+    Status construct_output_rowset_writer(uint32_t max_rows_per_segment, CompactionAlgorithm algorithm);
 
     Status check_version_continuity(const std::vector<RowsetSharedPtr>& rowsets);
     Status check_correctness(const Statistics& stats);
 
     // semaphore used to limit the concurrency of running compaction tasks
     static Semaphore _concurrency_sem;
 
+private:
```

Comment:
Place private declarations at the last.
https://google.github.io/styleguide/cppguide.html#Declaration_Order